custom-ingestion/Input.py

```diff
@@ -0,0 +1,30 @@
+# Create resource object as database.schema.table
+
+class Input:
+    input_type = 'dataset'  # only type needed for a first versions
+
+    def __init__(self,
+                 database: str,
+                 schema: str,
+                 table: str) -> object:
+        self.database = database
+        self.schema = schema
+        self.table = table
+        self.type = self.input_type
+
+    def __str__(self) -> str:
+        return f"Input type: {self.type} - '{self.table}' table, from '{self.schema}' 'schema and database '{self.database}'"
+
+    # Instance methods
+    def _get_path(self) -> str:
+        return '.'.join([self.database, self.schema, self.table])
+
+    def _get_input_entry(self) -> str:
+        return f"{{\"string\": \"urn:li:{self.type}:(urn:li:dataPlatform:prod,{self._get_path()},PROD)\"}}"
+
+    @staticmethod
+    def get_input_entries(inputs: list, snapshot_type: str) -> str:
```

Comment:
Although I do understand that by having this method here you're trying to associate the concern of listing the inputs to the class `Input` (and that this works), I don't agree with that decision, for 2 reasons.

- By doing this, you're somewhat embedding into Input logic that is completely external to this object. What I mean is that to the `Input` class, the only concerns should be its internal attributes and logic, but instead, we're creating logic that only makes sense when we consider how we're gonna use this object in `Snapshot` (or in a collection of these objects that we want to print like this).
- You're calling a normal method from a static method, which is a big no-no to me in most cases. IMHO the purpose of static methods is to include logic that solely concerns the class in which it was defined, but does not require an instance to execute (or is specifically intended not to be associated with an instance). Here, we're defining a static method that requires a collection of instances.

In my opinion, this should just be a method in `Snapshot`, since that's the true "owner" of this logic.
Alternatively, you can create a class that simply manages a collection of `Input` objects, but I think that's just overkill :D

Reply:
awesomely stated!!!
This is a really valid point and makes complete sense.

My first thought was to implement a `List Input` class, as we learn in college, but, as you said, it felt too verbose. My mind went directly to implement a static method to deal with objects of that same class.
Didn't even though occurred to me that `Input` was being called inside another object 😅 

What would be the best approach if it wasn't? create the referred `List Input` class? @joseppinto 
**NOTE:** I don't like the idea of defining this method in the main class.